server/src/main/java/org/elasticsearch/indices/IndicesWriteLoadStore.java

```diff
@@ -0,0 +1,201 @@
+        this(createBulkProcessor(settings, threadPool, client), clusterSettings, settings);
+    }
+
+    IndicesWriteLoadStore(BulkProcessor bulkProcessor, ClusterSettings clusterSettings, Settings settings) {
+        this.bulkProcessor = bulkProcessor;
+
+        this.enabled = ENABLED_SETTING.get(settings);
+        clusterSettings.addSettingsUpdateConsumer(ENABLED_SETTING, this::setEnabled);
+    }
+
+    static BulkProcessor createBulkProcessor(Settings settings, ThreadPool threadPool, Client client) {
+        return BulkProcessor.builder(client, new BulkProcessor.Listener() {
+            @Override
+            public void beforeBulk(long executionId, BulkRequest request) {}
+
+            @Override
+            public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
+                if (response.hasFailures()) {
+                    Map<String, String> failures = Arrays.stream(response.getItems())
+                        .filter(BulkItemResponse::isFailed)
+                        .collect(
+                            Collectors.toMap(
+                                BulkItemResponse::getId,
+                                BulkItemResponse::getFailureMessage,
+                                (msg1, msg2) -> Objects.equals(msg1, msg2) ? msg1 : msg1 + "," + msg2
+                            )
+                        );
+                    logger.warn("Failed to index some indices write load distributions: [{}]", failures);
+                }
+            }
+
+            @Override
+            public void afterBulk(long executionId, BulkRequest request, Throwable failure) {
+                long items = request.numberOfActions();
+                logger.warn(new ParameterizedMessage("Failed to index {} items into indices write load index", items), failure);
+            }
+        }, threadPool, threadPool, () -> {})
+            .setBulkActions(MAX_DOCUMENTS_PER_BULK_SETTING.get(settings))
+            .setBulkSize(MAX_BULK_SIZE_SETTING.get(settings))
+            .setConcurrentRequests(MAX_CONCURRENT_REQUESTS_SETTING.get(settings))
+            .setBackoffPolicy(BackoffPolicy.exponentialBackoff(TimeValue.timeValueMillis(1000), MAX_RETRIES_SETTING.get(settings)))
+            .build();
+    }
+
+    void putAsync(List<ShardWriteLoadDistribution> shardWriteLoadDistributions) {
+        if (enabled == false) {
+            return;
+        }
+
+        for (ShardWriteLoadDistribution shardWriteLoadDistribution : shardWriteLoadDistributions) {
+            try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+                shardWriteLoadDistribution.toXContent(builder, ToXContent.EMPTY_PARAMS);
+                final var request = new IndexRequest(INDICES_WRITE_LOAD_DATA_STREAM).source(builder).opType(DocWriteRequest.OpType.CREATE);
+                bulkProcessor.add(request);
```

Comment:
Will using the `BulkProcessor` here expose this at all to https://github.com/elastic/elasticsearch/issues/68468#issuecomment-772990141 ?